lib/sda/service.go

```diff
@@ -295,7 +297,7 @@ in place of the standard reply. The Client.Send method will catch that and retur
 
 // Client for a service
 type Client struct {
-	private abstract.Secret
+	Private abstract.Secret
```

Comment:
Ok from what I see, you use `client.Send` and `client.Private`. You need the `Private` because you sign something but the private key you use for *signing* and the private key you use for the *client part* (`Send`) *should / can* be separate; they have nothing in common. 
Normally the `Private` of a client is never actually used, since it's only given to `TCPHost` but never used - maybe it's something that has to be changed also ? - that was also the reason why to put it private, because a user of the `Client` api should **not care** about keys at all. If he wants to, then he embeds `Client` and use its own fabric.
Wouldn't it be possible to get the same behaviour with something like:
```
// Identity can both follow and update an IdentityList
type Identity struct {
	// client as unexported so it is not marshalled
	client *sda.Client
	// private key as exported so it is marshalled 
	// different role == different key
	Private abstract.Secret 
	ID         ID
	...
}
```
* Clear separation of role between signing and client request/response == different keys
* No need to export `Private` *just for that*

I'm sorry to bother about this but since we are doing an effort to have a clean API, I think it's important to think about those details. Moreover, it's only a suggestion, if any of you have a better idea, please !